llvm/lib/Transforms/Scalar/Scalarizer.cpp

```diff
@@ -1195,7 +1261,7 @@ bool ScalarizerVisitor::finish() {
     if (!Op->use_empty()) {
       // The value is still needed, so recreate it using a series of
       // insertelements and/or shufflevectors.
-      Value *Res;
+      Value *Res = nullptr;
```

Comment:
I think assigning null here may suppress a "not all paths initialize this variable" type warning if we were to add a code path that doesn't. Given that all of the code paths still do initialize the variable that might not be ideal.

Reply:
Reflexive change sorry. Uninitialized reads were a pattern I use to look for in security reviews. But looks like we never do anything but assign to Res so erroneous thinking on my part. Wasn't thinking this might be done this way to catch new conditions.